ebean-core/src/main/java/io/ebeaninternal/server/deploy/BeanProperty.java

```diff
@@ -759,12 +765,18 @@ public Object value(Object bean) {
     return getValueIntercept((EntityBean) bean);
   }
 
+  private boolean beanHasProperty(EntityBean bean) {
+    return descriptor.inheritInfo() == null // fast exit for non inherited beans
+      || owningType == null
+      || owningType.isInstance(bean);
+  }
+
   /**
    * Return the value of the property method.
    */
   public Object getValue(EntityBean bean) {
     try {
-      return getter.get(bean);
+      return beanHasProperty(bean) ? getter.get(bean) : null;
```

Comment:
Technically I don't think this is needed on `getValue()` but instead on `getValueIntercept()` only